yapapi/engine.py

```diff
@@ -459,12 +459,19 @@ async def _process_invoices(self) -> None:
 
     async def _process_debit_notes(self) -> None:
         """Process incoming debit notes."""
+        debit_note_processing_tasks = []
+        loop = asyncio.get_event_loop()
 
         async for debit_note in self._payment_api.incoming_debit_notes():
-            await self._process_debit_note(debit_note)
+            debit_note_processing_tasks.append(
+                loop.create_task(self._process_debit_note(debit_note))
+            )
             if self._payment_closing and not self._agreements_to_pay:
                 break
 
+        if debit_note_processing_tasks:
+            await asyncio.gather(*debit_note_processing_tasks)
```

Comment:
Will this handle a very large list of tasks? It's called after the computation is finished, the list contains all debit notes that were ever received.